File: instalacao-ci/application/models/Entity/ResultadosDoEncaminhamento.java
package br.edu.unipampa.app.domain;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity(name = "resultados_do_encaminhamento")
public class ResultadosDoEncaminhamento implements Comparable<ResultadosDoEncaminhamento> {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne(cascade = CascadeType.REFRESH)
    private Encaminhamento encaminhamento;

    @OneToMany(cascade = CascadeType.REFRESH, fetch = FetchType.LAZY, orphanRemoval = false)
    @JoinColumn(name = "resultados_do_encaminhamento_votantes", insertable = false, updatable = false)
    private List<Membro> votantes;

    private Integer quantidadeDeVotos = 0;

    private Double porcentagem = 0.0;

    public ResultadosDoEncaminhamento computaUmVoto() {
        setQuantidadeDeVotos();
        return this;
    }

    public ResultadosDoEncaminhamento eAdiciona(Membro votante) {
        if(votantes == null){
           votantes = new ArrayList<>();
        }
        this.votantes.add(votante);

        return this;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Encaminhamento getEncaminhamento() {
        return encaminhamento;
    }

    public void setEncaminhamento(Encaminhamento encaminhamento) {
        this.encaminhamento = encaminhamento;
    }

    public List<Membro> getVotantes() {
        return votantes;
    }

    public void setVotantes(List<Membro> votantes) {
        this.votantes = votantes;
    }

    public Integer getQuantidadeDeVotos() {
        return quantidadeDeVotos;
    }

    private void setQuantidadeDeVotos() {
        this.quantidadeDeVotos++;
    }

    public Double getPorcentagem() {
        return porcentagem;
    }

    public void setPorcentagem(Double porcentagem) {

        this.porcentagem = porcentagem;
    }

    public int compareTo(ResultadosDoEncaminhamento outroResultado){
        if (this.quantidadeDeVotos > outroResultado.quantidadeDeVotos) {
            return -1;
        }
        if (this.quantidadeDeVotos < outroResultado.quantidadeDeVotos){
            return 1;
        }
        return 0;
    }

    @Override
    public String toString() {
        return "ResultadosDoEncaminhamento{" +
                "id=" + id +
                ", quantidadeDeVotos=" + quantidadeDeVotos +
                ", porcentagem=" + porcentagem +
                '}';
    }



}

File: instalacao-ci/application/models/Entity/ResultadosDoEncaminhamento.php
<?php

namespace Entity;

use Doctrine\Common\Collections\ArrayCollection;


/**
 * @Entity
 * @Table(name="resultados_do_encaminhamento")
 */
class ResultadosDoEncaminhamento
{

    /**
     * @Id
     * @Column(type="integer", nullable=false)
     * @GeneratedValue(strategy="IDENTITY")
     */
    private $id;

    /**
     * @OneToOne(targetEntity="Encaminhamento", cascade={"refresh"})
     */
    private $encaminhamento;

    /**
     * @ManyToMany(targetEntity="Membro", orphanRemoval=false, cascade={"refresh"})
     * @JoinTable(name="resultados_do_encaminhamento_membros",
     *      joinColumns={@JoinColumn(name="resultados_do_encaminhamento_id", referencedColumnName="id")},
     *      inverseJoinColumns={@JoinColumn(name="membros_id", referencedColumnName="id", unique=true)}
     *      )
     */
    private $votantes;

    /**
     * @Column(type="integer", nullable=false)
     */
    private $quantidadeDeVotos = 0;

    /**
     * @Column(type="float", nullable=false)
     */
    private $porcentagem = 0.0;

    /**
     * ResultadosDoEncaminhamento constructor.
     */
    public function __construct()
    {
        $this->votantes = new ArrayCollection();
    }

    /**
     * @return mixed
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * @param mixed $id
     */
    public function setId($id)
    {
        $this->id = $id;
    }

    /**
     * @return mixed
     */
    public function getEncaminhamento()
    {
        return $this->encaminhamento;
    }

    /**
     * @param mixed $encaminhamento
     */
    public function setEncaminhamento($encaminhamento)
    {
        $this->encaminhamento = $encaminhamento;
    }

    /**
     * @return mixed
     */
    public function getVotantes()
    {
        return $this->votantes;
    }

    /**
     * @param mixed $votantes
     */
    public function setVotantes($votantes)
    {
        $this->votantes = $votantes;
    }

    /**
     * @return mixed
     */
    public function getQuantidadeDeVotos()
    {
        return $this->quantidadeDeVotos;
    }

    /**
     * @param mixed $quantidadeDeVotos
     */
    public function setQuantidadeDeVotos($quantidadeDeVotos)
    {
        $this->quantidadeDeVotos = $quantidadeDeVotos;
    }

    /**
     * @return mixed
     */
    public function getPorcentagem()
    {
        return $this->porcentagem;
    }

    /**
     * @param mixed $porcentagem
     */
    public function setPorcentagem($porcentagem)
    {
        $this->porcentagem = $porcentagem;
    }

//    public ResultadosDoEncaminhamento computaUmVoto() {
//        setQuantidadeDeVotos();
//        return this;
//    }
//
//    public ResultadosDoEncaminhamento eAdiciona(Membro votante) {
//        if(votantes == null){
//           votantes = new ArrayList<>();
//        }
//        this.votantes.add(votante);
//
//        return this;
//    }
//
//    public int compareTo(ResultadosDoEncaminhamento outroResultado){
//        if (this.quantidadeDeVotos > outroResultado.quantidadeDeVotos) {
//            return -1;
//        }
//        if (this.quantidadeDeVotos < outroResultado.quantidadeDeVotos){
//            return 1;
//        }
//        return 0;
//    }

}
